airflow/hooks/base_hook.py

```diff
@@ -48,3 +48,46 @@ def get_pandas_df(self, sql):
 
     def run(self, sql):
         raise NotImplemented()
+
+
+class BaseSqlHook(BaseHook):
+    """
```

Comment:
Sounds right. Unless the MySqlHook implementation works with other DBs. I think as look at it's vanilla SQL and DbApi it should be generalizable. PrestoHook should override it with `NotImplemented` though